lighthouse-core/fraggle-rock/gather/driver.js

```diff
@@ -68,20 +66,20 @@ class Driver {
 
   /** @return {Promise<void>} */
   async connect() {
-    if (this._session) return;
+    if (this.defaultSession !== throwingSession) return;
     const status = {msg: 'Connecting to browser', id: 'lh:driver:connect'};
     log.time(status);
     const session = await this._page.target().createCDPSession();
-    this._session = this.defaultSession = new ProtocolSession(session);
-    this._executionContext = new ExecutionContext(this._session);
-    this._fetcher = new Fetcher(this._session);
+    this.defaultSession = new ProtocolSession(session);
+    this._executionContext = new ExecutionContext(this.defaultSession);
+    this._fetcher = new Fetcher(this.defaultSession);
     log.timeEnd(status);
   }
 
   /** @return {Promise<void>} */
   async disconnect() {
-    if (!this._session) return;
-    await this._session.dispose();
+    if (this.defaultSession === throwingSession) return;
+    await this.defaultSession?.dispose();
```

Comment:
```suggestion
    await this.defaultSession.dispose();
```
I assume shouldn't be necessary because it's always defined?

Reply:
true. thanks for catching